### PR TITLE
Convert ReceiverRuntimeMetricsTests to unit tests and speed up some existing long sleep tests

### PR DESCRIPTION
**Pull request by christothes**
Just out of curiosity, I investigated the tests mentioned in #8892. Although the tests passed locally for me, it seemed that the mechanism being tested does not require a Live test. 

Scope of Work:
- Mocked out `EventHubClient`, `EventDataSender`, and `PartitionReceiver`
- Converted ReceiverRuntimeMetricsTests to unit tests
- Reduced some long sleeps in `ServiceFabricProcessor` tests

Before and after test run length (excludes Live tests):

Test run for C:\src\azure-sdk-for-net\artifacts\bin\Microsoft.Azure.EventHubs.Tests\Debug\netcoreapp2.1\Microsoft.Azure.EventHubs.Tests.dll(.NETCoreApp,Version=v2.1)
```
Test Run Successful.
Total tests: 71
     Passed: 41
    Skipped: 30
 Total time: 53.7411 Seconds
```
After:
```
Test Run Successful.
Total tests: 75
     Passed: 45
    Skipped: 30
 Total time: 8.7574 Seconds
```

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/ServiceFabricProcessor/TimeoutTests.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/ServiceFabricProcessor/TimeoutTests.cs]: nit:  Extra spaces

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/ServiceFabricProcessor/TimeoutTests.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/ServiceFabricProcessor/TimeoutTests.cs]: nit:  Extra spaces

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/ReceiverRuntimeMetricsTests.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/ReceiverRuntimeMetricsTests.cs]: 👏  Any time we can increase CI coverage and avoid live resources, its a win!

**Review by jsquire — APPROVED** [review]
LGTM.   Since this is the track one client, I'm going to ask that we wait for approval from at least one of the Event Hubs service team members before moving forward.

**Comment by jsquire** [comment]
@serkantkaraca, @JamesBirdsall, @sjkwak :   Would one of you be kind enough to review and offer your thoughts?

**Review by JamesBirdsall — APPROVED** [review]
This seems reasonable to me. Verifying that the service is correctly annotating outbound events is a service-side test. The client just needs to verify that it grabs the annotations and returns them accurately when GetPartitionRuntimeInformationAsync is called.

**Comment by jsquire** [comment]
Looks like we're in a good state and nothing else is pending.   @christothes:  If you would be so kind as to give a final thumbs up when you're ready, I'll merge this in.

**Comment by christothes** [comment]
@jsquire Ready when you are - thanks!

**Pull request merged**
